ui/app/controllers/oidc-mock.js

```diff
@@ -0,0 +1,32 @@
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import Ember from 'ember';
+
+export default class OidcMockController extends Controller {
+  @service router;
+
+  queryParams = ['auth_method', 'client_nonce', 'redirect_uri', 'meta'];
+
+  @action
+  signIn(fakeAccount) {
+    const url = `${this.redirect_uri.split('?')[0]}?code=${
+      fakeAccount.accessor
+    }&state=success`;
+    if (Ember.testing) {
+      this.router.transitionTo(url);
+    } else {
+      window.location = url;
+    }
+  }
+
+  @action
+  failToSignIn() {
+    const url = `${this.redirect_uri.split('?')[0]}?state=failure`;
+    if (Ember.testing) {
+      this.router.transitionTo(url);
+    } else {
+      window.location = url;
+    }
+  }
```

Comment:
`question`: This controller isn't going to make it into production, and only our test builds, no? Are we planning to refactor this later to no longer be a mock?

Reply:
As it stands, this controller/route *will* be in the production binary, but because of the condition in the Router, will only exist if we're in Mirage mode — that is, no running agent / customer would see it.

As such, the most important thing here is ease of internal testing for this route in particular.